packages/cozy-realtime/src/Socket.js

```diff
@@ -111,6 +112,18 @@ class Socket {
     }
   }
 
+  /**
+   * If Socket is not connected, connects.
+   * Does nothing if Socket is already connected
+   *
+   * @return {Promise} - Resolves when connected
+   */
+  async ensureConnected() {
+    if (!this.isOpen()) {
```

Comment:
If the socket is establishing the connection, this will start a new connection again. Maybe we should have a special case for `isConnecting()`